cardano-crypto-class/src/Cardano/Crypto/Hash/Packed.hs

```diff
@@ -0,0 +1,66 @@
+{-# LANGUAGE DeriveGeneric #-}
+module Cardano.Crypto.Hash.Packed where
+
+import qualified Data.Primitive.ByteArray as BA
+import qualified Data.ByteString.Short as SBS
+import           Data.ByteString.Short (ShortByteString)
+import qualified Data.ByteString.Short.Internal as SBSI
+import           NoThunks.Class (NoThunks)
+import           GHC.Generics (Generic)
+import           Data.Word (Word64, Word8)
+import qualified Control.Monad.ST as ST
+import           Control.DeepSeq (NFData)
+
+data Bytes32 = Bytes32
+    {-# UNPACK #-} !Word8 -- length
+    {-# UNPACK #-} !Word64
+    {-# UNPACK #-} !Word64
+    {-# UNPACK #-} !Word64
+    {-# UNPACK #-} !Word64
+  deriving (Eq, Generic)
+
+
+instance Ord Bytes32 where
+  -- Note: the derived ord instance would produce different orderings
+  -- on machines with different byte orders.
+  compare x y = compare (unpack x) (unpack y)
```

Comment:
This is a huge problem. It will affect performance tremendously, therefore this approach ins not acceptable!!!
Consider that on each insert into a `Map` or a `Set` there will be 2*log(n) conversions to SBS in order to perform a comparison.

Reply:
What do you suggest?

Machine-dependent ordering isn't acceptable either.

Reply:
Implementing a fast machine independent ordering without conversion to SBS. This will be a major pain, but it is doable